compiler/rustc_errors/src/emitter.rs

```diff
@@ -2260,13 +2260,18 @@ impl HumanEmitter {
                     buffer.puts(*row_num, max_line_num_len + 1, "+ ", Style::Addition);
                 }
                 [] => {
-                    draw_col_separator(buffer, *row_num, max_line_num_len + 1);
+                    draw_col_separator_no_space(buffer, *row_num, max_line_num_len + 1);
                 }
                 _ => {
                     buffer.puts(*row_num, max_line_num_len + 1, "~ ", Style::Addition);
                 }
             }
-            buffer.append(*row_num, &normalize_whitespace(line_to_add), Style::NoStyle);
+            buffer.puts(
```

Comment:
What's the difference here (specifically, why do we need `+ 3`)?  Would be worth adding comments explaining a lot of the logic for the `+ N`s in this function.

Reply:
Added explanation for that changed part.